eZ/Publish/Core/Persistence/Cache/UserHandler.php

```diff
@@ -32,6 +32,10 @@ public function create(User $user)
 
         // Clear corresponding content cache as creation of the User changes it's external data
         $this->cache->invalidateTags(['content-fields-' . $user->id]);
+        $this->cache->deleteItems([
+            'ez-user-' . str_replace('@', '§', $user->login) . '-by-login',
+            'ez-user-' . str_replace('@', '§', $user->email) . '-by-email',
+        ]);
```

Comment:
Hmm, shouldn't we consider tagging by e-mail and login also instead of deleting items? Not sure if this makes a difference though.
POV ping @andrerom

Reply:
Actually, what you propose is the second way to fix this, both work. 

If I would like to do it like you propose, then I would need to add here:
```
$userByLogin = $this->cache->getItem('ez-user-' . str_replace('@', '§', $user->login) . '-by-login');
$userByLogin->set($user);
$this->cache->save($userByLogin);
        
$userByEmail = $this->cache->getItem('ez-user-' . str_replace('@', '§', $user->email) . '-by-email');
$userByEmail->set($user);
$this->cache->save($userByEmail);
```
and I'm not sure whether we want to do it in this method. 

Personally, I prefer to stay with the current approach, but it's up to @alongosz and @andrerom.

Reply:
What @alongosz is suggesting to tagging by idenifiers, we could but as it's unique there is no need.

What you are suggesting @kmadejski is to pre-warm cache instead of clearing, with 2.0 (kernel 7.0) I on purpose omitted that to 1. keep things simple (logic, testing, responsibilities, ..) and 2. since it was buggy and sometimes placed things in cache which would never be used.